codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/endpoint/Util.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.client.smithy.endpoint
+
+import software.amazon.smithy.codegen.core.Symbol
+import software.amazon.smithy.rulesengine.language.syntax.Identifier
+import software.amazon.smithy.rulesengine.language.syntax.parameters.Parameter
+import software.amazon.smithy.rulesengine.language.syntax.parameters.ParameterType
+import software.amazon.smithy.rulesengine.traits.ContextParamTrait
+import software.amazon.smithy.rust.codegen.core.rustlang.RustReservedWords
+import software.amazon.smithy.rust.codegen.core.rustlang.RustType
+import software.amazon.smithy.rust.codegen.core.smithy.makeOptional
+import software.amazon.smithy.rust.codegen.core.smithy.rustType
+import software.amazon.smithy.rust.codegen.core.util.letIf
+import software.amazon.smithy.rust.codegen.core.util.toSnakeCase
+
+/**
+ * Utility function to convert an [Identifier] into a valid Rust identifier (snake case)
+ */
+fun Identifier.rustName(): String {
+    return this.toString().stringToRustName()
+}
+
+fun String.stringToRustName(): String = RustReservedWords.escapeIfNeeded(this.toSnakeCase())
```

Comment:
These utils appear to be geared towards endpoints given the module, but IntelliJ will encourage people to import this function everywhere in `codegen-client`. This seems too broad of an extension function to me.

Reply:
good catch—made private / refactored